opencog/rule-engine/Rule.cc

```diff
@@ -486,23 +486,13 @@ std::string Rule::to_string(const std::string& indent) const
 
 bool Rule::has_name_capture() const
 {
-	HandleSeq vardecls = VariableListCast((this->get_vardecl()))
-			->get_variables().varseq;
 	HandleSeq boundvars = BindLinkCast(this->get_rule())->get_variables()
 			.varseq;
-	size_t sb = boundvars.size();
-	size_t sd = vardecls.size();
+	FreeVariables fv = VariableListCast((this->get_vardecl()))->get_variables();
```

Comment:
Could you change the brace the following way:
 ```FreeVariables fv = (VariableListCast(this->get_vardecl()))->get_variables();``` just to make it less ambiguous.